cutil/parser/httpfileresource/httpresource.go

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpresource
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+)
+
+// HTTPResource could hold explicit information for HTTP calls
+// right now just helps us for type enforcment
+type HTTPResource struct{}
+
+// NewHTTPResource creates a new file resource fetcher
+func NewHTTPResource(fr HTTPResource) *HTTPResource {
+	return &fr
+}
+
+// Read reads file from a http url
+func (fr *HTTPResource) Read(sourcePath string) ([]byte, error) {
+	// After checking https://www.ietf.org/rfc/rfc3986.txt turns out
+	// that almost everything is a valid URL
+	targetURL, err := url.ParseRequestURI(sourcePath)
+	if err != nil {
+		return []byte(""), err
+	}
+
+	// Here we check if the url starts with http or https
+	match, err := regexp.MatchString("[http|https]", targetURL.Scheme)
```

Comment:
ParseRequestURI will return error if HTTP(s) is missing, we don't need this check here.
Look here - 
https://play.golang.org/p/THOCNhqY-2